conn.go

```diff
@@ -32,18 +32,131 @@ import (
 // Commands are buffered. Flush sends all buffered commands in a single batch.
 type Conn struct {
 	TestDial nltest.Func // for testing only; passed to nltest.Dial
-	NetNS    int         // Network namespace netlink will interact with.
-	sync.Mutex
+	NetNS    int         // fd referencing the network namespace netlink will interact with.
+
+	lasting  bool       // establish a lasting connection to be used across multiple netlink operations.
+	mu       sync.Mutex // protects the following state
 	messages []netlink.Message
 	err      error
+	nlconn   *netlink.Conn // netlink socket using NETLINK_NETFILTER protocol.
+}
+
+// ConnOption is an option to change the behavior of the nftables Conn returned by Open.
+type ConnOption func(*Conn)
+
+// New returns a netlink connection for querying and modifying nftables. Some
+// aspects of the new netlink connection can be configured using the options
+// WithNetNSFd, WithTestDial, and AsLasting.
+//
+// A lasting netlink connection should be closed by calling Close() to close the
+// underlying netlink connection and release the associated resources and
+// pending operations.
```

Comment:
Is “release the […] pending operations” really the best verb? It will cancel any pending operations, right? Maybe we should use “cancel” as verb to clarify this.